### ISSUE TITLE
How to seed `community_detection`

### ISSUE THREAD

**Issue by thechargedneutron** [UKPLab/sentence-transformers#1859]
Thanks for the package.

If we run `examples/applications/clustering/fast_clustering.py` multiple times the generated communities are different each time. How can we modify the code to generate seeded communities? 

I tried adding seed from different packages but they do not seem to work:
```
import transformers
import random
import numpy as np
import torch

seed_value = 42
random.seed(seed_value)
np.random.seed(seed_value)
torch.manual_seed(seed_value)
torch.cuda.manual_seed_all(seed_value)
transformers.set_seed(seed_value)
```

**Comment by thechargedneutron** [comment]
Okay, the non-determinism comes from the use of `set()` when constructing the sentences. It is confusing since the conversion from set to list does not preserve the order (well, by design). I am closing this issue now. If the maintainers think it might be a good idea to shift to dictionary for determinism, I can submit a PR.
